src/sentry_core.c

```diff
@@ -848,4 +850,108 @@ sentry_set_span(sentry_value_t transaction)
         scope->span = transaction;
     }
 }
+
+sentry_value_t
+sentry_span_start_child(
+    sentry_value_t parent, char *operation, char *description)
+{
+    size_t max_spans = SENTRY_SPANS_MAX;
+    SENTRY_WITH_OPTIONS (options) {
+        max_spans = options->max_spans;
+    }
+
+    if (sentry_value_is_null(parent)) {
+        SENTRY_DEBUG("no transaction available to create a child under");
+        goto fail;
+    }
```

Comment:
You were noting in chat that the example stopped working at some point. This is the reason why. In the example your first child span is not linked up to the parent transaction.